testutils/src/main/java/com/microsoft/identity/internal/testutils/CacheUtils.java

```diff
@@ -0,0 +1,159 @@
+        SharedPreferences sharedPref = getSharedPreferences(SharedPrefName);
+        SharedPreferences.Editor prefEditor = sharedPref.edit();
+        Map<String, ?> cacheEntries = sharedPref.getAll();
+
+        //get all the key from the cache entry, verify and edit it.
+        for (Map.Entry<String, ?> cacheEntry : cacheEntries.entrySet()) {
+            String keyToEdit = cacheEntry.getKey();
+            if (predicate.test(keyToEdit)) {
+                Object cacheValue = cacheEntries.get(keyToEdit);
+                if (cacheValue instanceof String) {
+                    Credential credential = CACHE_KEY_VALUE_DELEGATE.fromCacheValue((String) cacheValue, Credential.class);
+                    credential.setSecret(editor.apply(credential.getSecret()));
+                    prefEditor.putString(keyToEdit, CACHE_KEY_VALUE_DELEGATE.generateCacheValue(credential));
+                    prefEditor.apply();
+                }
+            }
+        }
+    }
+
+    /**
+     * utility function to edit Access Token.
+     *
+     * @param SharedPrefName Name of the shared preference.
+     * @param editor         Functional interface for token editing method.
+     */
+    public void editAllAccessTokenInCache(@NonNull final String SharedPrefName, Function<String, String> editor) {
+        Predicate<String> predicate = new Predicate<String>() {
+            @Override
+            public boolean test(String cacheKey) {
+                return isAccessToken(cacheKey);
+            }
+        };
+
+        editAllTokenInCache(SharedPrefName, predicate, editor);
+    }
+
+    /**
+     * utility function to edit Refresh Token.
+     *
+     * @param SharedPrefName Name of the shared preference.
+     * @param editor         Functional interface for token editing method.
+     */
+    public void editAllRefreshTokenInCache(@NonNull final String SharedPrefName, Function<String, String> editor) {
+        Predicate<String> predicate = new Predicate<String>() {
+            @Override
+            public boolean test(String cacheKey) {
+                return isRefreshToken(cacheKey);
+            }
+        };
+
+        editAllTokenInCache(SharedPrefName, predicate, editor);
+    }
+
+    private static SharedPreferences getSharedPreferences(final String sharedPrefName) {
+        final Context context = ApplicationProvider.getApplicationContext();
+        return context.getSharedPreferences(sharedPrefName, Context.MODE_PRIVATE);
+    }
+
+    /**
+     * Split the string into 3 segment separated by period "." and set the
+     * random character at the random position in the 3rd segment.
+     *
+     * @param string string to be edited.
+     * @return edited string.
+     */
+    private static String editTokenSignature(String string) {
+        String[] segments = string.split(".");
+        String signature = segments[segments.length - 1];
```

Comment:
Do you want to add a check if '.' is not found and 3 segments are not there? Put Assert there

Reply:
added AssertionError exception whenever string is not segmented into 3 parts by period. ".".